src/plugin/yjsEditor.ts

```diff
@@ -60,40 +51,51 @@ export const YjsEditor = {
     fn();
 
     if (!wasRemote) {
-      Promise.resolve().then(() => IS_REMOTE.delete(editor));
+      IS_REMOTE.delete(editor);
     }
   },
+};
 
-  /**
-   * Apply Yjs events to slate
-   */
-  applyYjsEvents: (editor: YjsEditor, events: Y.YEvent[]): void => {
-    YjsEditor.asRemote(editor, () => {
-      applyYjsEvents(editor, events);
-    });
-  },
+function localOperations(editor: YjsEditor): Set<Operation> {
+  const operations = LOCAL_OPERATIONS.get(editor);
+  invariant(operations, 'YjsEditor without attached local operations');
+  return operations;
+}
 
-  /**
-   * Performs an action as a local operation.
-   */
-  asLocal: (editor: YjsEditor, fn: () => void): void => {
-    const wasLocal = YjsEditor.isLocal(editor);
-    IS_LOCAL.add(editor);
+function trackLocalOperations(editor: YjsEditor, operation: Operation): void {
+  if (!YjsEditor.isRemote(editor)) {
+    localOperations(editor).add(operation);
+  }
+}
 
-    fn();
+/**
+ * Applies a slate operations to the bound shared type.
+ */
+function applyLocalOperations(editor: YjsEditor): void {
+  const editorLocalOperations = localOperations(editor);
 
-    if (!wasLocal) {
-      IS_LOCAL.delete(editor);
-    }
-  },
+  applySlateOps(
+    YjsEditor.sharedType(editor),
+    Array.from(editorLocalOperations),
+    slateYjs
+  );
 
-  /**
-   * Returns whether the editor currently is applying a remove change to the yjs doc.
-   */
-  isLocal: (editor: YjsEditor): boolean => {
-    return IS_LOCAL.has(editor);
-  },
-};
+  editorLocalOperations.clear();
```

Comment:
```suggestion
```

Reply:
See WeakSet changes above

Reply:
Thinking about it some more: Never mind that, we should keep the clear here anyway